From 0ae39c1f172638849a1b5bc2be0592b3388a3670 Mon Sep 17 00:00:00 2001
Message-Id: <0ae39c1f172638849a1b5bc2be0592b3388a3670.1632746837.git.stefan@agner.ch>
In-Reply-To: <9cb97076d98f7f68534abb3d1f596644ae730841.1632746837.git.stefan@agner.ch>
References: <9cb97076d98f7f68534abb3d1f596644ae730841.1632746837.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 23 Sep 2021 23:52:44 +0200
Subject: [PATCH 7/9] nvme: improve readability of nvme_setup_prps()

Improve readability by introducing consts, reuse consts where
appropriate and adding variables with discriptive name.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 drivers/nvme/nvme.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/nvme/nvme.c b/drivers/nvme/nvme.c
index c61dab20c5..56d5e9f38f 100644
--- a/drivers/nvme/nvme.c
+++ b/drivers/nvme/nvme.c
@@ -76,12 +76,12 @@ static int nvme_wait_ready(struct nvme_dev *dev, bool enabled)
 static int nvme_setup_prps(struct nvme_dev *dev, u64 *prp2,
 			   int total_len, u64 dma_addr)
 {
-	u32 page_size = dev->page_size;
+	const u32 page_size = dev->page_size;
+	const u32 prps_per_page = (page_size >> 3) - 1;
 	int offset = dma_addr & (page_size - 1);
 	u64 *prp_pool;
 	int length = total_len;
 	int i, nprps;
-	u32 prps_per_page = (page_size >> 3) - 1;
 	u32 num_pages;
 
 	length -= (page_size - offset);
@@ -119,9 +119,9 @@ static int nvme_setup_prps(struct nvme_dev *dev, u64 *prp2,
 	prp_pool = dev->prp_pool;
 	i = 0;
 	while (nprps) {
-		if (i == ((page_size >> 3) - 1)) {
-			*(prp_pool + i) = cpu_to_le64((ulong)prp_pool +
-					page_size);
+		if (i == prps_per_page) {
+			u64 next_prp_list = (u64)prp_pool + page_size;
+			*(prp_pool + i) = cpu_to_le64(next_prp_list);
 			i = 0;
 			prp_pool += page_size;
 		}
-- 
2.33.0

